src/components/markdownRenderer/index.tsx

```diff
@@ -0,0 +1,50 @@
+import ReactMarkdown from "react-markdown";
+import React from "react";
+import style from "./markdown-renderer.module.css";
+import ReactPlayer from "react-player";
+
+interface MarkdownRendererProps {
+    children: string;
+}
+
+export const MarkdownRenderer: React.FC<MarkdownRendererProps> = (props) => {
+    return (
+        <>
+            {props.children.split("\n").map((line) => (
+                <>
+                    {line === "" ? (
+                        <div className="newline" />
+                    ) : (
+                        <BaseRenderer content={line} />
+                    )}
+                </>
+            ))}
+        </>
+    );
+};
+
+const BaseRenderer: React.FC<{ content: string }> = ({ content }) => (
+    <div className="md-row">
+        <ReactMarkdown
+            components={{
+                a: ({ node, className, children, ...rest }) => (
+                    <>
+                        {content.includes("$[") ? (
+                            <>
+                                <div>
+                                    <ReactPlayer controls url={rest.href} />
+                                </div>
+                            </>
+                        ) : (
+                            <a href={rest.href}>{children}</a>
+                        )}
+                    </>
+                ),
+            }}
+            className={style.markdown + " right"}
+        >
+            {content.replace("$[", "[")}
+        </ReactMarkdown>
+        {content}
```

Comment:
Is this necessary or was it just in for debugging?

also, do we need the `md-row` div wrapping each of the lines? We could maybe just have a div wrapping the entire markdown output, but I think each ReactMarkdown line already has its own div

Reply:
1) U right mb
2) Basically I had to put each MD row on its own div with {flex-direction: row} to make sure the styling of components on each line was correct even with $ to indent, while also laying out the components on a given line left to right. I think we may have to keep it even tho it's not the most elegant solution